klog_test.go

```diff
@@ -1243,6 +1243,74 @@ func TestLogFilter(t *testing.T) {
 	}
 }
 
+func TestLogLevelWithLogrAddedFields(t *testing.T) {
+	logger := new(testLogr)
+
+	_, file, _, ok := runtime.Caller(0)
+	if !ok {
+		t.Errorf("Unable to get source file info")
+	}
+	thisFile := filepath.Base(file)
+
+	testCases := []struct {
+		name     string
+		logFunc  func(args ...interface{})
+		expected testLogrEntry
+	}{{
+		name:    "Info",
+		logFunc: Info,
+		expected: testLogrEntry{
+			severity:      infoLog,
+			msg:           "foo\n",
```

Comment:
This retains the `Info/Warning/Error` behavior of adding a newline even though the log line becomes a structured log message, happy to change this around so the log message is trimmed if that seems more suitable